autocorpus/bioc/collection.py

```diff
@@ -4,22 +4,38 @@
 the collection to a dictionary representation.
 """
 
+from __future__ import annotations
+
 import xml.etree.ElementTree as ET
+from dataclasses import dataclass, field
 from typing import Any
 
 from .document import BioCDocument
 
 
+@dataclass
 class BioCCollection:
+    """A class representing a BioC collection."""
+
     def __init__(
         self,
-        source: str = "",
-        date: str = "",
-        key: str = "",
-        documents: list[BioCDocument] | None = None,
-        infons: dict[str, str] | None = None,
-        version: str = "",
+        source: str = field(default_factory=str),
```

Comment:
This isn't quite right. You should do away with defining an `__init__` method entirely and just define the parameters at the class level. There are quite a few others that have been done this way and should be updated.

For reference: The `BioCPassage` one has been done correctly.